Sources/Apollo/RequestBodyCreator.swift

```diff
@@ -39,16 +36,11 @@ extension RequestBodyCreator {
       body["variables"] = variables.jsonEncodableObject
     }
 
-    if sendOperationIdentifiers {
-      guard let operationIdentifier = operation.operationIdentifier else {
-        preconditionFailure("To send operation identifiers, Apollo types must be generated with operationIdentifiers")
-      }
-
-      body["id"] = operationIdentifier
-    }
-
     if sendQueryDocument {
-      body["query"] = operation.queryDocument
+      guard let document = operation.definition?.queryDocument else {
+        preconditionFailure("To send query documents, Apollo types must be generated with `OperationDefinition`s.")
```

Comment:
why `preconditionFailure` instead of just `fatalError`, out of curiosity?

Reply:
This is actually a failure based on the precondition of you telling the code gen configuration that they should generate the operation definitions. This isn't a regular run time failure based on bad inputs. The way to fix this is to change your config and re-run Code Generation.